code/espurna/garland.cpp

```diff
@@ -0,0 +1,223 @@
+/*
+GARLAND MODULE
+Copyright (C) 2020 by Dmitry Blinov <dblinov76 at gmail dot com>
+
+Tested on 60 led strip. 
+!!! For more leds can cause WDT rebot. Need to be carefully tested for more than 60 leds !!!
+The most time consuming operation is actually showing leds by Adafruit Neopixel. It take about 1870 mcs.
+More long strip can take more time to show. 
+Currently animation calculation, brightness calculation/transition and showing makes in one loop cycle.
+Debug output shows timings. Overal timing should be not more that 3000 ms.
+
+For longer strips have sense to separate show operation from animation/brightness calculation. Or even
+divide showing for parts and process them in different cycles.
+*/
+
+#include "garland.h"
+
+#if GARLAND_SUPPORT
+
+#include <Adafruit_NeoPixel.h>
+
+#include <vector>
+
+#include "garland/anims.h"
+#include "garland/color.h"
+#include "garland/palette.h"
+#include "garland/scene.h"
+#include "web.h"
+#include "ws.h"
+
+const char* NAME_GARLAND_ENABLED        = "garlandEnabled";
+const char* NAME_GARLAND_BRIGHTNESS     = "garlandBrightness";
+
+const char* NAME_GARLAND_SWITCH         = "garland_switch";
+const char* NAME_GARLAND_SET_BRIGHTNESS = "garland_set_brightness";
+
+constexpr int LEDS                      = GARLAND_LEDS;
+constexpr int PIN                       = GARLAND_D_PIN; // WS2812 pin number
+
+#define EFFECT_UPDATE_INTERVAL_MIN      5000  // 5 sec
+#define EFFECT_UPDATE_INTERVAL_MAX      10000 // 10 sec
+
+bool _garland_enabled                   = true;
+unsigned long _last_update              = 0;
+unsigned long _interval_effect_update;
+
+int paletteInd;
+int animInd                             = 0; 
```

Comment:
'static' in loop?

Reply:
There is no class for garland (think it is redundant) I store "state": current used animation and palette in "static". Would it be better to define them static int inside the loop?

Reply:
Yes, the reason being nothing else uses these but the loop function